### ISSUE TITLE
First time opening Character Preference TGUI freezes the server and can cause runtimes

### ISSUE THREAD

**Issue by larentoun** [tgstation/tgstation#70309]
## Reproduction:

How to replicate:
1. Launch a server (as usual)
2. Open character preferences UI. It will lag the server as a whole (for me at least, it lags for 10-ish seconds). After the lag, the UI will still be blank-white
3. Join the game/Press ready and start the game
4. Wait for the previous blank-white UI to finally process and give runtimes. The runtimes are there, since its trying to get a client of lobby-mob, which doesn't have a client anymore, since we have moved to our playable living mob.

Note: the lag only happens when char-pref UI opens for the first time. New calls to open it (on the same client or others) will not cause a lag and will load their UI in reasonable time, not allowing to join the game before the UI loads.
Note 2: the blank-white UI will properly load in aprox. 20-ish seconds, IF you don't leave the lobby screen.

**Comment by Mothblocks** [comment]
I don't get any runtimes. The slower load time might be from @LemonInTheDark 's lazy asset changes?

**Comment by LemonInTheDark** [comment]
Yep, this is "intended" in that if we violate the cache we need to build it all at once.
Honestly I've been considering trying to keep the old cache on hand, and slowly replace it with new assets. not sure how deranged that is.

**Comment by Mothblocks** [comment]
Not sure what you mean

**Comment by Mothblocks** [comment]
It is absolutely not reasonable to check which icons are going to change and it's only reasonable to rebuild the entire cache

Does this happen every time or just the first time the assets need to build?

**Comment by LemonInTheDark** [comment]
This happens only the first time, every time on local to my annoyance.
I don't mean checking only the icons that fail. I mean holding onto the old cache, and slowly replacing it rather then doing it in bulk.
Although now that I say that, that might be kinda pointless since most ui assets are in one file so it'd just be use the old cache until the new one's made.
That said, I do want to make this less of an issue. It bothers me that it happens in the first place

**Comment by LemonInTheDark** [comment]
Honestly I am a bit unhappy with the current lazylaoded asset system in general, and would like to make it less hacky and a bit less slow to generate after initialization

**Comment by Mooshimi** [comment]
> Wait for the previous blank-white UI to finally process and give runtimes.

Can we see the runtimes

**Comment by larentoun** [comment]
> > Wait for the previous blank-white UI to finally process and give runtimes.
> 
> Can we see the runtimes

![image](https://user-images.githubusercontent.com/31931237/196290489-2842493b-c77f-475c-aaa7-9e180ccb666d.png)

**Comment by Mooshimi** [comment]
> > > Wait for the previous blank-white UI to finally process and give runtimes.
> > 
> > 
> > Can we see the runtimes
> 
> ![image](https://user-images.githubusercontent.com/31931237/196290489-2842493b-c77f-475c-aaa7-9e180ccb666d.png)

tyty

**Comment by LemonInTheDark** [comment]
I believe what happens in that runtime is the client disconnects, and so is unable to open the menu

**Comment by larentoun** [comment]
> I believe what happens in that runtime is the client disconnects, and so is unable to open the menu

Not really. It is stated in the issue report, that when you open CharPrefs UI, it doesn't load immediately. If you launch the game, you are changing mobs. The white-blank UI will eventually load and will try to access the mob which you had in lobby, which doesn't have a client anymore
